src/include/homestore/index/index_table.hpp

```diff
@@ -235,12 +240,18 @@ class IndexTable : public IndexTableBase, public Btree< K, V > {
         }
 
         auto& root_buf = static_cast< IndexBtreeNode* >(new_root.get())->m_idx_buf;
-        wb_cache().transact_bufs(ordinal(), m_sb_buffer, root_buf, {}, {}, r_cast< CPContext* >(context));
+        IndexBufferPtrList freed_bufs;
+        if (freed_root) {
+            freed_bufs.push_back(static_cast<IndexBtreeNode *>(freed_root.get())->m_idx_buf);
+        }
+        // Meta is similar to a leftmost child here - it should always be the up buffer for (both) root(s)
+        wb_cache().transact_bufs(ordinal(), nullptr, m_sb_buffer, {root_buf}, freed_bufs,
```

Comment:
what would be the problem of meta to be parent? would you please elaborate here?

Reply:
I removed this change since it may not be that necessary and I cannot recall the detailed case it was for.
We can go back to this later when we find more corner cases regarding the root node.